cuelang: refactor

AndersonTorres opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres commented:
@aaronjheng please review

wegank commented on pkgs/by-name/cu/cuelang/tests/001-all-good.nix:
```diff
@@ -0,0 +1,17 @@
+, cue
+}:
+
+stdenvNoCC.mkDerivation {
```
I think `runCommand` would be preferable to `stdenvNoCC` with `buildCommand`?

AndersonTorres replied:
I never tried this. Let's see...

AndersonTorres replied:
Ugly, but it worked.

wegank commented on pkgs/top-level/all-packages.nix:
```diff
@@ -3678,9 +3678,8 @@ with pkgs;
-  cue = callPackage ../development/tools/cue { };
-
-  writeCueValidator = callPackage ../development/tools/cue/validator.nix { };
+  cue = cuelang;
```
I don't think it's necessary to rename it to `cuelang`, few distros have that as an attribute name. https://repology.org/project/cuelang/versions

AndersonTorres replied:
I was trying to package Cue, a music player: https://github.com/ravachol/cue
Here is it: https://github.com/NixOS/nixpkgs/pull/249095

That being said, I think it can be useful to differentiate them even being rarely used.

aaronjheng replied:
`cue` has been renamed to `kew`. See https://github.com/ravachol/kew/commit/2334207e83f68d9fd781997be5db4c410ef4f865.

AndersonTorres replied:
OK, "reverting" the rename.

AndersonTorres committed:
cue: internalize writeCueValidator via passthru

A small step to by-name migration.

AndersonTorres committed:
cue: move custom installCheckPhase to passthru.tests

As recommended by the Nixpkgs documentation, custom tests are better served by
passthru.tests, whilst installCheckPhase is recommended for tests that come from
the upstream.

AndersonTorres committed:
cue: migrate to by-name
